apache/couchdb-fauxton
# Remove dependency with Backbone.Model

## Overview

Introduces new classes `Model` and `Collection` to directly replace Backbone's `Model` and `Collection`. The new classes implement the main Backbone's methods (e.g. `save()` and  `destroy()`) using `fetch`. This provides a quick(er) way to remove the dependency with Backbone since most of the Fauxton code/logic that relies on Backbone models and collections can remain unchanged.

Summary of changes:
 - replaced all references to Backbone.Model/Collection with Model/Collection
 - it now uses `class extends Model` / `class extends Collection` instead of `Backbone.Model.extend` / `Backbone.Collection.extend`
 - callers now use `then/catch` (Promises) instead of `done/fail` (jQuery deferred).

 - Misc:
   - Replaced `Fauxton.VersionInfo` with a `fetch` call.
   - Removed unused classes: `BulkDeleteDocModel`, `BulkDeleteDocCollection`, `MangoBulkDeleteDocCollection`


## Testing recommendations

- Verify you can create, list, delete databases
- Verify you can list all docs, list view results
- Edit and save docs
- Edit and save views

## GitHub issue number


## Checklist
